Experiment with the Slack API to prove the status can be programmatically updated

- [x] Create Slack app (https://api.slack.com/apps/A035D5C5LBS)
- [x] Configure permissions (scopes) in https://api.slack.com/apps/A035D5C5LBS/oauth?
  -  `users:read` scope (https://api.slack.com/scopes/users:read)
⚠️ Both `Bot Token Scopes` and `User Token Scopes` have been set to `users:read` (https://api.slack.com/scopes/users:read) because I do not know the difference
  - `users.profile:read` scope (https://api.slack.com/scopes/users.profile:read)
⚠️ Both `Bot Token Scopes` and `User Token Scopes` have been set to `users.profile:read` scope (https://api.slack.com/scopes/users.profile:read) because I do not know the difference
  - `users:write` scope (https://api.slack.com/scopes/users:write)
  - `users.profile:write` scope (https://api.slack.com/scopes/users.profile:write)
  ⚠️ This scope (`users.profile:write`) only exists for users, it DOES NOT exist for bots!
  - `dnd:read` scope (https://api.slack.com/scopes/dnd:read)
  - `dnd:write` scope (https://api.slack.com/scopes/dnd:write)
  ⚠️ This scope (`dnd:write`) only exists for users, it DOES NOT exist for bots!
  
- [x] Test
  - [x] `users.getPresence`
  - [x] `users.setPresence` 
  - [x] `users.profile.get`
  - [x] `users.profile.set`
  - [x] `dnd.info`
  - [x] `dnd.setSnooze`
  
- [ ] Export the manifest

https://api.slack.com/methods/users.setPresence/test

### User Token, wrong status
`https://slack.com/api/users.setPresence?presence=abc&pretty=1

Authorization: Bearer xoxp-*

{
    "ok": false,
    "error": "invalid_arguments",
    "response_metadata": {
        "messages": [
            "[ERROR] must be a valid enum value [json-pointer:/presence]"
        ]
    }
}`

### User Token
`https://slack.com/api/users.setPresence?presence=away&pretty=1

Authorization: Bearer xoxp-*

{
    "ok": true
}`

`https://slack.com/api/users.setPresence?presence=auto&pretty=1

Authorization: Bearer xoxp-*

{
    "ok": true
}`

### Bot Token
⚠️ IT IS NOT POSSIBLE TO PROVIDE A USERID AS PARAMETER
`https://slack.com/api/users.setPresence?presence=auto&pretty=1

Authorization: Bearer xoxb-*

{
    "ok": true
}`

`https://slack.com/api/users.setPresence?presence=away&pretty=1

Authorization: Bearer xoxb-*

{
    "ok": true
}`

--- a/manifest.json
+++ b/manifest.json
@@ -1,0 +1,37 @@
+{
+    "display_information": {
+        "name": "(What's the Status)",
+        "description": "Effortless (Slack) presence management for a healthy work-life balance",
+        "background_color": "#2c2d30",
+        "long_description": "`(What's the Status)` aims to support a healthy work-life balance, by tackling your digital presence (Slack, MS Teams, Zoom, Google, GitHub, LinkedIn, email, phone, ... anything!)\r\n\r\nIt starts with Slack ... no one knows what's next.\r\n\r\nEffortless[1]: Make It Easier to Do What Matters Most"
+    },
+    "features": {
+        "bot_user": {
+            "display_name": "(What's the Status)",
+            "always_online": false
+        }
+    },
+    "oauth_config": {
+        "scopes": {
+            "user": [
+                "dnd:read",
+                "users.profile:read",
+                "users.profile:write",
+                "users:read",
+                "users:write",
+                "dnd:write"
+            ],
+            "bot": [
+                "users.profile:read",
+                "users:read",
+                "users:write",
+                "dnd:read"
+            ]
+        }
+    },
+    "settings": {
+        "org_deploy_enabled": false,
+        "socket_mode_enabled": false,
+        "token_rotation_enabled": false
+    }
+}

--- a/manifest.yml
+++ b/manifest.yml
@@ -1,0 +1,35 @@
+display_information:
+  name: (What's the Status)
+  description: Effortless (Slack) presence management for a healthy work-life balance
+  background_color: "#2c2d30"
+  long_description: "`(What's the Status)` aims to support a healthy work-life balance, by tackling your digital presence (Slack, MS Teams, Zoom, Google, GitHub, LinkedIn, email, phone, ... anything!)\r
+
+    \r
+
+    It starts with Slack ... no one knows what's next.\r
+
+    \r
+
+    Effortless[1]: Make It Easier to Do What Matters Most"
+features:
+  bot_user:
+    display_name: (What's the Status)
+    always_online: false
+oauth_config:
+  scopes:
+    user:
+      - dnd:read
+      - users.profile:read
+      - users.profile:write
+      - users:read
+      - users:write
+      - dnd:write
+    bot:
+      - users.profile:read
+      - users:read
+      - users:write
+      - dnd:read
+settings:
+  org_deploy_enabled: false
+  socket_mode_enabled: false
+  token_rotation_enabled: false

